File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        
        if(head == null)
            return head;
        ListNode p = head, itr = head.next;
        
        while(itr != null){
            if(head.val != itr.val){
                head.next = itr;
                head = head.next;
            }
            itr = itr.next;
        }
        head.next = null;
        
        return p;
        
    }
}

File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        
        if head == None:
            return head
        
        tmp, itr = head, head.next
        while itr != None:
            if head.val != itr.val:
                head.next = itr
                head = head.next
                
            itr = itr.next
        head.next = None
                
        return tmp
            
        
        
        
        